### PR TITLE
Quick fixes for usability

### PR DESCRIPTION
**Pull request by kenherring**
A handful few small fixes here that I think improve usability for those times when we're not working with a single project, or just want to run some quick code.

If for whatever reason you think we need to split up this PR just let me know... happy to do so.

# `openedge-project.json` .config

* Don't require that `"dbconnections": []` be defined in `openedge-project.json`
  * Fixes this runtime error:
    image
* See #107 ~~Allow for using `openedge-project.jsonc` instead of `openedge-project.json`.  Most of the other VSCode config files allow for comments, and it would be really helpful for this plugin too.~~
* When no source or propath entries are provided in `openedge-project.json` default the propath to `'.'`
* If no `"oeversion"` key is given, use the default from user settings if configured.

# `launch.json` config

* When creating a debug configuration the port field is set to "string" when it should really be an integer as it fails the schema validation.
    image
    
# Terminal related fixes

* Create the `.builder/` directory on the first run of a task/command if needed to fix this error:
    image
* When using the "**ABL: Run with _progres**" command change the dirsep character to `/` so it plays nicely with Git Bash.  Seems that using the forward slash is compatible with everything, but the back slash is not so friendly.  Tested with Git Bash, cmd, PowerShell.  No change needed on the other commands.
    image

### CODE REVIEW

**Comment by gquerret** [comment]
Regarding `openedge-project.json`, I agree on all points except jsonc. In fact, I don't disagree, it's just that the json file is also read from the Java language server, and there was no support for jsonc in the Gson library. I'll have a look. The LS needs to be modified to handle those default values in the same way.
And another point is that it's currently possible to specify `default: true` on multiple OE versions (in vscode config). Not sure if there's a way to enforce only one.

* `launch.json`: not sure if service name are resolved, but agree that an integer is easier

**Comment by gquerret** [comment]
Will merge the PR when changes are done in the LS

**Comment by gquerret** [comment]
Looks like Gson already has support for comments in Json.

File: resources/abl-src/dynrun.p [hunk 1]
```diff
@@ -42,28 +42,41 @@ ASSIGN configJson = CAST(jsonParser:ParseFile(SESSION:PARAMETER), JsonObject).
...
-  IF (dbEntry:has("name") AND dbEntry:has("connect")) THEN DO:
-    LOG-MANAGER:WRITE-MESSAGE(SUBSTITUTE("Connecting to DB '&1': '&2'", dbEntry:GetCharacter("name"), dbEntry:GetCharacter("connect"))).
-    CONNECT VALUE(dbEntry:GetCharacter("connect")) NO-ERROR.
-    IF ERROR-STATUS:ERROR THEN DO:
-      IF (ERROR-STATUS:NUM-MESSAGES > 1) OR (ERROR-STATUS:GET-NUMBER(1) NE 1552) THEN DO:
-        LOG-MANAGER:WRITE-MESSAGE(SUBSTITUTE("Unable to connect to '&1'" , dbEntry:GetCharacter("name"))).
-        DO i = 1 TO ERROR-STATUS:NUM-MESSAGES:
-          LOG-MANAGER:WRITE-MESSAGE(ERROR-STATUS:GET-MESSAGE(i)).
+//DB connections + aliases
+DO ON ERROR UNDO, LEAVE:
```
Comment (gquerret) [inline resources/abl-src/dynrun.p:46]: Wouldn't configJson:has("databases") be better ?

**Comment by cverbiest** [comment]
Port was changed from string to integer.
Personally I always use integers for port but AFAIK you can still use port names

File: resources/abl-src/dynrun.p (hunk 1 above)
Comment (kenherring) [reply resources/abl-src/dynrun.p]: Most certainly better!  Updated.

**Comment by kenherring** [comment]
> * And another point is that it's currently possible to specify `default: true` on multiple OE versions (in vscode config). Not sure if there's a way to enforce only one.

What do you think about adding a new user config for the default instead?  Maybe this:

```jsonc
    "abl.configuration.runtimes.default": "12.2",
    "abl.configuration.runtimes": [
        {
            "name": "12.2",
            "path": "C:/Progress/Openedge"
        },
        {
            "name": "11.7",
            "path": "C:\\Progress\\OpenEdge-11.7"
        }
    ],

**Comment by gquerret** [comment]
Implementation is ready in LS for openedge-project.json changes. Can you also squash and rebase against develop (and not main) ?

**Comment by kenherring** [comment]
closing in place of #108

**Pull request closed without merging**